ufo2fontir/src/source.rs

```diff
@@ -157,34 +179,38 @@ impl Source for DesignSpaceIrSource {
 
         // UFO filename => map of layer
         let mut layer_cache = HashMap::new();
-
         let mut glif_locations: HashMap<PathBuf, Vec<DesignSpaceLocation>> = HashMap::new();
+        let mut glyph_order = Vec::new();
 
-        for source in designspace.sources {
+        for source in designspace.sources.iter() {
             // Track files within each UFO
             // The UFO dir *must* exist since we were able to find fontinfo in it earlier
             let ufo_dir = self.designspace_dir.join(&source.filename);
 
             let location = to_ir_location(&source.location);
 
-            for (glyph_name, glif_file) in glif_files(&ufo_dir, &mut layer_cache, &source)? {
+            for (glyph_name, glif_file) in glif_files(&ufo_dir, &mut layer_cache, source)? {
                 if !glif_file.exists() {
                     return Err(Error::FileExpected(glif_file));
                 }
                 let glif_file = glif_file.clone();
-                glyphs
-                    .entry(glyph_name)
-                    .or_default()
-                    .track_file(&glif_file)?;
+                let entry = glyphs.entry(glyph_name.clone());
+                if let hash_map::Entry::Vacant(_) = entry {
+                    glyph_order.push(glyph_name);
```

Comment:
the glyph order is not the order in which you find glyphs listed in a UFO layer's contents.plist, the latter is a dictionary and as such unsorted (the fact that you may see it ordered alphabetically is simply a way to serialise the plist dict in a deterministic way).

UFO defines a "public.glyphOrder" key in the UFO lib.plist, which is a list of glyph name strings. That's what fontmake (actually ufo2ft) uses. If it is not defined, ufo2ft sorts the glyphs alphabetically while ensuring that if .notdef is present (which usually is) it is placed in the first (gid 0) position:

https://github.com/googlefonts/ufo2ft/blob/dd738cdcddf61cce2a744d1cafab5c9b33e92dd4/Lib/ufo2ft/util.py#L23-L45

since you want the glyph order of the variable font (right now you're assuming there's 1 VF per designspace file, which isn't the case for v5 but anyway), then you have to get the public.glyphOrder from the default master (the source that is at the default location on all the axes), which is the starting point for the variable font to be built.

Reply:
for reference: https://unifiedfontobject.org/versions/ufo3/lib.plist/#publicglyphorder

note that public.glyphOrder may contain more or even less glyphs than are in the font.. 😅 
you want to prune glyphs that are not actually in the font, and also append at the end the glyphs that are in the font but are not explicitly listed in the public.glyphOrder

basically what ufo2ft's makeOfficialGlyphOrder does

Reply:
v5 designspace I definitely want to tackle separately :)

Reply:
I have added mk1 of building the glyph order. Until we do #22 - seems like that needs to be soon - the way it locates the default master will be entirely wrong.

Reply:
Also made .glyphs a bit more similar to ufo, though still honoring file order as default.